### ISSUE TITLE
Warn against exact floating-point comparsions.

### ISSUE THREAD

**Issue by zvavybir** [rust-lang/rust-clippy#8583]
### What it does

Warns against usage of exact comparison operators (`==` and `!=`, not things like `>` or `<=`) since floating point numbers rarely are exactly equal.

Where I got the idea from: [https://gforth.org/manual/Floating-Point-Tutorial.html](https://gforth.org/manual/Floating-Point-Tutorial.html)
Prior art: [https://users.rust-lang.org/t/comparing-floats-for-equality/54523](https://users.rust-lang.org/t/comparing-floats-for-equality/54523) and [https://bitbashing.io/comparing-floats.html](https://bitbashing.io/comparing-floats.html)

### Lint Name

exact-floating-point

### Category

pedantic

### Advantage

- Avoids subtle errors

### Drawbacks

There might be occasions where the programmer knows that the value is exact (e.g. `inf` or `nan` should be common – although they should be handled by dedicated functions).

### Example

```rust
let a = 0.1f64;
let b = 0.2f64;
if a + b == 0.3
{
  println!("This isn't reachable!");
}
```

Could be written as:

```rust
let a = 0.1f64;
let b = 0.2;
if (a + b - 0.3).abs() < 0.001*((a+b).abs()+0.3f64.abs())
{
  println!("This is reachable!");
}
```

**Comment by xFrednet** [comment]
Hey @zvavybir, thank you for the issue. This sounds like [`clippy::float_cmp`](https://rust-lang.github.io/rust-clippy/master/index.html#float_cmp) and [`clippy::float_cmp_const`](https://rust-lang.github.io/rust-clippy/master/index.html#float_cmp_const) respectively. Is this what you were looking for? :upside_down_face:

**Comment by zvavybir** [comment]
Yes, it is :) ! I forgot to check with pedantic enabled :(
